### PR TITLE
GOV-29: Fixing NuGet publishing failing due to GitHub runners now having NuGet.org registered as a package source

### PR DESCRIPTION
**Pull request by Piedone**
[GOV-29](https://lombiq.atlassian.net/browse/GOV-29)

[GOV-29]: https://lombiq.atlassian.net/browse/GOV-29?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Piedone** [comment]
You might need this @Psichorex.

File: .github/actions/publish-nuget/Add-SourceLinkPackage.ps1 [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
+{
+    Register-PackageSource -Name NuGet.org -Location https://api.nuget.org/v3/index.json -ProviderName NuGet
+}
+else
+{
+    Write-Output 'Package source for NuGet.org is already registered.'
+}
 
 # Get the latest version of the package.
 $latestPackage = Find-Package -Name Microsoft.SourceLink.GitHub -Source NuGet.org | Sort-Object Version -Descending | Select-Object -First 1
```
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/actions/publish-nuget/Add-SourceLinkPackage.ps1:14]: If any of these fail then we need the whole workflow fail, as it happens currently.

File: .github/actions/publish-nuget/Add-SourceLinkPackage.ps1 [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+# Register NuGet.org as a package source since GitHub runners don't necessarily have it by default.
+$existingSource = Get-PackageSource -Name NuGet.org -ErrorAction SilentlyContinue
+
+if (-not $existingSource)
+{
+    Register-PackageSource -Name NuGet.org -Location https://api.nuget.org/v3/index.json -ProviderName NuGet
+}
+else
+{
+    Write-Output 'Package source for NuGet.org is already registered.'
```
Comment (dministro) [inline .github/actions/publish-nuget/Add-SourceLinkPackage.ps1:10]: https://github.com/Lombiq/UI-Testing-Toolbox/actions/runs/7823210350/job/21343751208#step:4:105

**Review by dministro — APPROVED** [review]

File: .github/actions/publish-nuget/Add-SourceLinkPackage.ps1 (hunk 2 above)
Comment (Piedone) [reply .github/actions/publish-nuget/Add-SourceLinkPackage.ps1:10]: What do you mean?

Comment (dministro) [reply .github/actions/publish-nuget/Add-SourceLinkPackage.ps1:10]: I just noted it myself.

Comment (Piedone) [reply .github/actions/publish-nuget/Add-SourceLinkPackage.ps1:10]: Ah OK.

**Review by dministro — APPROVED** [review]

**Pull request merged**